### PR TITLE
Remove workaround channel listener in ESQL

### PR DESCRIPTION
**Pull request by dnhatn**
Relates #100528

### CODE REVIEW

**Comment by dnhatn** [comment]
@elasticmachine test this please

**Comment by dnhatn** [comment]
test this please

**Comment by dnhatn** [comment]
run elasticsearch-ci/part-3

**Comment by alex-spies** [comment]
Closes https://github.com/elastic/elasticsearch/issues/102667

**Review by original-brownbear — APPROVED** [review]
LGTM!

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeService.java [hunk 1]
```diff
@@ -502,7 +502,7 @@ public void messageReceived(DataNodeRequest request, TransportChannel channel, T
             final var sessionId = request.sessionId();
             final var exchangeSink = exchangeService.getSinkHandler(sessionId);
             parentTask.addListener(() -> exchangeService.finishSinkHandler(sessionId, new TaskCancelledException("task cancelled")));
-            final ActionListener<DataNodeResponse> listener = new OwningChannelActionListener<>(channel);
```
Comment (alex-spies) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeService.java:505]: Simillarly here, this is probably fine but technically now `DataNodeResponse`s never get released, I think.

**Review by alex-spies — APPROVED** [review]
Just minor remarks - but this can be merged as-is IMO!

**Review by ChrisHegarty — APPROVED** [review]

**Comment by dnhatn** [comment]
Thanks everyone!

**Pull request merged**